packages/sdk-effects/iframe-manager/src/lib/iframe-manager.effects.ts

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2025 Ping Identity Corporation. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+/* eslint-disable @typescript-eslint/no-empty-function */
+import { AuthorizeUrl } from '@forgerock/sdk-types';
+
+// Define specific options for the iframe request
+export interface IframeRequestOptions {
+  url: string;
+  timeout: number;
+  // Specify query parameters expected on success/error if needed,
+  // otherwise, parse all parameters found.
+  // Example: expectedParams?: string[];
+}
+
+type ResolvedParams<T> = Record<string, T>;
+
+type Noop = () => void;
+
+/**
+ * Initializes the Iframe Manager effect.
+ * This follows the functional effect pattern, returning an API
+ * within a closure. Configuration could be passed here if needed
+ * for default behaviors (e.g., default timeout), but currently,
+ * all config is per-request.
+ *
+ * @returns An object containing the API for managing iframe requests.
+ */
+export default function iframeManagerInit(/*config: OAuthConfig*/) {
+  /**
+   * Creates an iframe to make a background request to the specified URL,
+   * waits for a redirect, and resolves with the query parameters from the
+   * redirect URL.
+   *
+   * @param options - The options for the iframe request (URL, timeout).
+   * @returns A Promise that resolves with the query parameters from the redirect URL or rejects on timeout or error.
+   */
+  const getParamsFromIFrame = (options: {
+    urlParams: string[];
+    url: AuthorizeUrl;
+    requestTimeout: number;
+  }): Promise<ResolvedParams<string>> => {
+    const { url, requestTimeout, urlParams } = options;
+
+    return new Promise((resolve, reject) => {
+      let iframe: HTMLIFrameElement | null = document.createElement('iframe');
+      let timerId: number | ReturnType<typeof setTimeout> | null = null;
+
+      // Define these within the promise closure to avoid retaining
+      // references after completion.
+      let onLoadHandler: () => void = () => {};
+      let cleanup: Noop = () => {};
+
+      cleanup = (): void => {
+        if (timerId) {
+          clearTimeout(timerId);
+          timerId = null;
+        }
+        if (iframe) {
+          iframe.removeEventListener('load', onLoadHandler);
+          // Check if iframe is still mounted before removing
+          if (iframe.parentNode) {
+            iframe.remove();
+          }
+          iframe = null; // Dereference iframe for garbage collection
+        }
+        onLoadHandler = () => {};
+        cleanup = () => {};
+      };
+
+      onLoadHandler = (): void => {
```

Comment:
I _think_ the thing we're missing here is waiting for either a success or error param before resolving/rejecting. If I'm reading this right, we are assuming success or failure upon the first redirect. But, this may not always be true. There may be additional redirects on the server-side before the redirect back to the client app.

That's what this was for in the legacy code: https://github.com/ForgeRock/forgerock-javascript-sdk/blob/develop/packages/javascript-sdk/src/oauth2-client/index.ts#L127. So, there's a condition saying, "now that the iframe has loaded a page, does it contain the appropriate success or error param? If not, keep listening for more onload events."